src/lib/storage/table.hpp

```diff
@@ -36,17 +41,18 @@ class Table : private Noncopyable {
   uint64_t row_count() const;
 
   // returns the number of chunks (cannot exceed ChunkID (uint32_t))
-  ChunkID chunk_count() const;
+  ChunkCount chunk_count() const;
 
   // returns the chunk with the given id
   Chunk& get_chunk(ChunkID chunk_id);
   const Chunk& get_chunk(ChunkID chunk_id) const;
 
   // Adds a chunk to the table. If the first chunk is empty, it is replaced.
-  void emplace_chunk(Chunk chunk);
+  // This method intentionally takes the unique ownership of the chunk.
+  void emplace_chunk(std::unique_ptr<Chunk> chunk);
 
   // Returns a list of all column names.
-  const std::vector<std::string>& column_names() const;
+  const std::vector<std::string> column_names() const;
```

Comment:
The public interface was not supposed to be changed. But we can see that you had to do it due to your self-defined `Column` data type.